src/core/message-parsing/presentAssistantMessage.ts

```diff
@@ -149,7 +170,15 @@ export async function presentAssistantMessage(cline: Task) {
 			await cline.say("text", content, undefined, block.partial)
 			break
 		}
-		case "tool_use":
+		case "log_message": {
```

Comment:
This looks like a resurrection of #3165 (which was closed), but this pull request is a refactor in should not introduce any new features. 

I have no problem introducing the new feature if it gets approved, but I do not think it's a good idea to sneak in features as part of large refactors, such a feature should be introduced as a separate pull request after this refactor is merged.

Reply:
I don't disagree in theory.  I started out to just update my Logging branch and decided the whole thing needed refactoring.  So in theory your right but in practice I deviated from theory :)

Reply:
And turns out the Logging really helped reveal edge cases in the parsing.